File: leet-code/704. Binary Search/BinarySearch704_2.java
package com.example;

public class BinarySearch704_2 {
    public int search(int[] nums, int target) {
        return binarySearch(nums, 0, nums.length - 1, target);
    }

    private int binarySearch(int[] nums, int left, int right, int target) {
        if (left > right) {
            return -1;
        }

        int mid = left + (right - left) / 2;
        if (nums[mid] == target) {
            return mid;
        }

        if (nums[mid] > target) {
            return binarySearch(nums, left, mid - 1, target);
        }
        return binarySearch(nums, mid + 1, right, target);
    }
}

File: leet-code/704. Binary Search/BinarySearch704_2.kt
package com.example

class BinarySearch704_2 {
    fun search(nums: IntArray, target: Int): Int {
        var low = 0
        var high = nums.size - 1
        while (low < high) {
            val mid = low + (high - low) / 2
            if (nums[mid] < target) {
                low = mid + 1
            } else {
                high = mid
            }
        }
        return if (nums[low] == target) low else -1
    }
}
